test/hummingbot/strategy/avellaneda_market_making/test_avellaneda_market_making.py

```diff
@@ -0,0 +1,1145 @@
+
+    def test_base_asset(self):
+        self.assertEqual(self.trading_pair.split("-")[0], self.strategy.base_asset)
+
+    def test_quote_asset(self):
+        self.assertEqual(self.trading_pair.split("-")[1], self.strategy.quote_asset)
+
+    def test_trading_pair(self):
+        self.assertEqual(self.trading_pair, self.strategy.trading_pair)
+
+    def test_get_price(self):
+        # Avellaneda Strategy get_price is simply a wrapper for MarketTradingPairTuple.get_mid_price()
+        self.assertEqual(self.market_info.get_mid_price(), self.strategy.get_price())
+
+    def test_get_last_price(self):
+        # TODO: Determine if the get_last_price() function is needed in Avellaneda Strategy
+        # Note: MarketTrradingPairTuple does not have a get_last_price() function
+
+        # self.assertEqual(self.market_info.get_last_price(), self.strategy.get_last_price())
+        pass
+
+    def test_get_mid_price(self):
+        self.assertEqual(self.market_info.get_mid_price(), self.strategy.get_mid_price())
+
+    def test_market_info_to_active_orders(self):
+        order_tracker = self.strategy.order_tracker
+
+        self.assertEqual(order_tracker.market_pair_to_active_orders, self.strategy.market_info_to_active_orders)
+
+        # Simulate order being placed
+        limit_order: LimitOrder = LimitOrder(client_order_id="test",
+                                             trading_pair=self.trading_pair,
+                                             is_buy=True,
+                                             base_currency=self.trading_pair.split("-")[0],
+                                             quote_currency=self.trading_pair.split("-")[1],
+                                             price=Decimal("101.0"),
+                                             quantity=Decimal("10"))
+
+        self.simulate_place_limit_order(self.strategy, self.market_info, limit_order)
+
+        self.assertEqual(1, len(self.strategy.market_info_to_active_orders))
+        self.assertEqual(order_tracker.market_pair_to_active_orders, self.strategy.market_info_to_active_orders)
+
+    def test_active_orders(self):
+        self.assertEqual(0, len(self.strategy.active_orders))
+
+        # Simulate order being placed
+        limit_order: LimitOrder = LimitOrder(client_order_id="test",
+                                             trading_pair=self.trading_pair,
+                                             is_buy=True,
+                                             base_currency=self.trading_pair.split("-")[0],
+                                             quote_currency=self.trading_pair.split("-")[1],
+                                             price=Decimal("101.0"),
+                                             quantity=Decimal("10"))
+
+        self.simulate_place_limit_order(self.strategy, self.market_info, limit_order)
+
+        self.assertEqual(1, len(self.strategy.active_orders))
```

Comment:
The validation should include checking that the active_order present is the one that was just simulated, maybe comparing the order id.

Reply:
The `BacktestMarket` class creates a limit order as follows:
`<buy/sell>://<trading_pair>/<random_uuid>`
But yes I agree that I should include a better assertion check for this.